src/styles/main.scss

```diff
@@ -1,3 +1,56 @@
+@import "./utils/functions";
+@import "./utils/variables.scss";
+
 body {
   margin: 0;
+  padding: 30px;
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px;
+  width: getLimitSize($day-box-size , $boxGap);
+  padding: 10px;
+  margin: 0 auto;
+
+  @each $day , $margin in $days {
+    &--start-day--#{$day} :first-child {
+      margin-left: $margin;
+    }
+  }
+
+  @for $lastDay from 28 to 31 {
+    &--month-length--#{$lastDay} :nth-child(n + #{$lastDay+1}) {
+      display: none;
+    }
+  }
+
+  &__box {
+    position: relative;
+    width: $day-box-size;
+    height: $day-box-size;
+    box-sizing: border-box;
+    background-color: #eee;
+    border: 1px solid black;
+    @for $index from 1 through 31 {
+      &:nth-child(#{$index})::before {
+        content: "#{$index}";
+        position: absolute;
+        top: 50%;
+        right: 50%;
+        transform: translate(50% , -50%);
+        display: block;
+        font-family: Arial, Helvetica, sans-serif;
+        font-size: 30px;
+      }
+    }
+    &:hover {
+      transform: translateY(-20px);
+      background-color: pink;
+      transition: all 0.5s;
```

Comment:
move transition to &__box to prevent jump after unhover

Reply:
fixed